frontend/src/app/shared/components/file-links/file-link-list/file-link-list.component.ts

```diff
@@ -198,82 +180,83 @@ export class FileLinkListComponent extends UntilDestroyedMixin implements OnInit
+    const fileLinks = this.resource.fileLinks as unknown&{ href:string };
+    return `${fileLinks.href}?filters=[{"storage":{"operator":"=","values":["${this.storage.id}"]}}]`;
   }
 
-  private prepareInformationBox(header:string, content:string, icon:string):void {
-    this.informationBoxHeader = header;
-    this.informationBoxContent = content;
-    this.informationBoxIcon = icon;
+  private setAuthorizationCallbackCookie(nonce:string):void {
+    this.cookieService.set(`oauth_state_${nonce}`, window.location.href, {
+      path: '/',
+    });
+  }
+
+  private static authorizationFailureActionUrl(baseUrl:string, nonce:string):string {
+    return `${baseUrl}&state=${nonce}`;
   }
 
   private initializeLocales():void {
     const storageType = this.storageTypeMap[this.storage._links.type.href];
     this.text = {
       infoBox: {
-        emptyStorageHeader: this.i18n.t('js.label_link_files_in_storage', { storageType }),
-        emptyStorageContent: this.i18n.t('js.label_no_file_links', { storageType }),
-        emptyStorageButton: this.i18n.t('js.label_open_storage', { storageType }),
-        connectionErrorHeader: this.i18n.t('js.label_no_storage_connection', { storageType }),
-        connectionErrorContent: this.i18n.t('js.label_storage_connection_error', { storageType }),
-        authorizationFailureHeader: this.i18n.t('js.label_login_to_storage', { storageType }),
-        authorizationFailureContent: this.i18n.t('js.label_storage_not_connected', { storageType }),
-        loginButton: this.i18n.t('js.label_storage_login', { storageType }),
+        emptyStorageHeader: this.i18n.t('js.storages.link_files_in_storage', { storageType }),
+        emptyStorageContent: this.i18n.t('js.storages.information.no_file_links', { storageType }),
+        emptyStorageButton: this.i18n.t('js.storages.open_storage', { storageType }),
+        fileLinkErrorHeader: this.i18n.t('js.storages.information.live_data_error'),
+        fileLinkErrorContent: this.i18n.t('js.storages.information.live_data_error_description', { storageType }),
+        connectionErrorHeader: this.i18n.t('js.storages.no_connection', { storageType }),
+        connectionErrorContent: this.i18n.t('js.storages.information.connection_error', { storageType }),
+        authorizationFailureHeader: this.i18n.t('js.storages.login_to', { storageType }),
+        authorizationFailureContent: this.i18n.t('js.storages.information.not_logged_in', { storageType }),
+        loginButton: this.i18n.t('js.storages.login', { storageType }),
       },
       actions: {
-        linkFile: this.i18n.t('js.label_link_files_in_storage', { storageType }),
+        linkFile: this.i18n.t('js.storages.link_files_in_storage', { storageType }),
       },
     };
   }
 
   private initializeStorageTypes() {
-    this.storageTypeMap[nextcloud] = this.i18n.t('js.label_nextcloud');
+    this.storageTypeMap[nextcloud] = this.i18n.t('js.storages.types.nextcloud');
   }
```

Comment:
Is it really necessary to initialise these strings by hand? I thought, that angular was clever enough to use the correct  variable value once the string is used. But I might be wrong on this..

So something like 
```
public text = {
  loginButton: (storageType:string) => this.I18n.t(''js.storages.login', { storageType }),
}
```

and then `this.text.infoBox.loginButton(this.storageType)`

Reply:
refactored to functions